src/order/directed.lean

```diff
@@ -189,6 +214,27 @@ lemma is_bot_iff_is_min [is_directed α (≥)] : is_bot a ↔ is_min a :=
 
 lemma is_top_iff_is_max [is_directed α (≤)] : is_top a ↔ is_max a := ⟨is_top.is_max, is_max.is_top⟩
 
+/-- The property that a function preserves `is_lub` on directed sets. -/
+def preserves_lub_on_directed [preorder β] (f : α → β) : Prop :=
+∀ (d : set α) (a : α), d.nonempty → directed_on (≤) d → is_lub d a → is_lub (f '' d) (f a)
```

Comment:
Can you move this new definition to a new section at the end of the file?

Reply:
I've created a new section `scott_continuous` at the end of the file.